dash/include/dash/algorithm/Sort.h

```diff
@@ -0,0 +1,1177 @@
+        auto    src_begin = it_g_nlt_nle + lidx;
+
+        dart_handle_t handle;
+        dash::internal::get_handle(
+            src_begin.dart_gptr(),
+            l_nlt_nle_data + lidx,
+            NLT_NLE_BLOCK,
+            &handle);
+        return handle;
+      });
+
+  DASH_ASSERT_EQ(
+      handles.size(),
+      p_unit_info.valid_remote_partitions.size(),
+      "invalid size of handles vector");
+
+  auto const last_p   = p_unit_info.valid_remote_partitions.back();
+  auto       out_last = l_nlt_nle_data + last_p * NLT_NLE_BLOCK;
+
+  dash::Future<value_t*> fut_result(
+      // wait
+      [out_last, &handles]() {
+        // Wait for all get requests to complete:
+        auto _out = out_last;
+        DASH_LOG_TRACE(
+            "dash::detail::psort_validate_partitions [Future]()",
+            "  wait for",
+            handles.size(),
+            "async get request");
+        DASH_LOG_TRACE(
+            "dash::detail::psort_validate_partitions [Future]",
+            "  _out:",
+            _out);
+        if (handles.size() > 0) {
+          if (dart_waitall_local(handles.data(), handles.size()) != DART_OK) {
+            DASH_LOG_ERROR(
+                "dash::detail::psort_validate_partitions [Future]",
+                "  dart_waitall_local failed");
+            DASH_THROW(
+                dash::exception::RuntimeError,
+                "dash::detail::psort_validate_partitions [Future]: "
+                "dart_waitall_local failed");
+          }
+        }
+        else {
+          DASH_LOG_TRACE(
+              "dash::detail::psort_validate_partitions [Future]",
+              "  No pending handles");
+        }
+        DASH_LOG_TRACE(
+            "dash::detail::psort_validate_partitions [Future] >",
+            "  async requests completed, _out:",
+            _out);
+        return _out;
+      });
+
+  // copy local portion
+  index_t lidx      = myid * NLT_NLE_BLOCK;
+  auto    cpy_begin = it_g_nlt_nle + lidx;
+  auto    lbegin    = cpy_begin.local();
+  std::copy(lbegin, lbegin + NLT_NLE_BLOCK, l_nlt_nle_data + lidx);
+
+  // wait for all copies
+  fut_result.wait();
+
+  auto const& acc_partition_count = p_unit_info.acc_partition_count;
+
+  for (auto const& border_idx : valid_partitions) {
+    auto const p_left  = p_borders.left_partition[border_idx];
+    auto const nlt_idx = p_left * NLT_NLE_BLOCK;
+
+    auto const peer_idx = p_left + 1;
+
+    if (l_nlt_nle[nlt_idx] < acc_partition_count[peer_idx] &&
+        acc_partition_count[peer_idx] <= l_nlt_nle[nlt_idx + 1]) {
+      p_borders.is_stable[border_idx] = true;
+    }
+    else {
+      if (l_nlt_nle[nlt_idx] >= acc_partition_count[peer_idx]) {
+        p_borders.upper_bound[border_idx] = partitions[border_idx];
+      }
+      else {
+        p_borders.lower_bound[border_idx] = partitions[border_idx];
+      }
+    }
+  }
+  for (auto& handle : handles) {
+    DASH_ASSERT_RETURNS(DART_OK, dart_handle_free(&handle));
```

Comment:
Not necessary, `dart_waitall_local` will free the handles before returning

Reply:
OK, right. I have simply "copied" it since it is a result of the API changes with futures. Its is modified now.